guess-the-country/country-frontend/src/App.vue

```diff
@@ -126,6 +122,22 @@ export default {
         }
       },
     },
+    getNextButtonCondition() {
+      let next_button_condition = this.sequence_mode === 'classic'&&(this.explanation || (this.control && this.prediction_city)) && this.round < 11
+              || this.sequence_mode==='recommender' && this.user_city_answer && this.round < 11
+              || this.sequence_mode==='basic' && this.user_city_answer
+      return next_button_condition
+    },
+    get_WhatAIGuess_ButtonCondition_treatment() {
+      let ai_guess_button_condition_treatment = this.sequence_mode==='classic'&&!this.prediction_city && this.user_city_answer && !this.control && this.round != 11
+              ||this.sequence_mode==='recommender'&& !this.prediction_city && !this.control && this.round != 11
+      return ai_guess_button_condition_treatment
+    },
+    get_WhatAIGuess_ButtonCondition_control() {
+      let ai_guess_button_condition_control = this.sequence_mode==='classic' && !this.prediction_city && this.user_city_answer && this.control && this.round != 11
+              ||this.sequence_mode==='recommender'&& !this.prediction_city && this.control && this.round != 11
+      return ai_guess_button_condition_control
+    },
```

Comment:
Wenn ich das richtig sehe, sind die bis auf `this.control` identisch?

Ich würde das so lösen, damit es übersichtlicher und verständlicher ist, was hier eigentlich vor sich geht und der Sinn ist:

```javascript
showAIGuessButton() {
  if (this.sequence.mode === 'classic') {
    return this.prediction_city && this.user_city_answer && this.round != 11
  } else if (this.sequence.mode === 'recommender') {
    return !this.prediction_city && this.round != 11
}
```

und bei den buttons dann

```html
<button v-if="showAIGuessButton && !this.control" ... />
```

Es geht mir einfach darum, dass wir hier keine endlose, schwer zu lesende Verkettung von Variablen haben, sondern dass man auf einen Blick sieht, was los ist: "zeige den Button wenn der User nicht in der Kontrollgruppe ist"

Reply:
@ionicsolutions ah, ich glaube ich hab verstanden was du meinst, danke für die Erklärung! Hab alles angepasst und ich finde es schon viel schöner! 

Das einzig unschöne ist noch, dass ich bei jeder computed Funktion ein else { return false } hinten anhängen musste, das gar nichts macht, aber ohne beschwert er sich dass er einen Rückgabewert braucht... sollen wir das so lassen, oder gibts da eine elegantere Lösung? 
Ich hätte halt die else if Bedingungen mit dem jeweiligen sequence mode schon gern so gelassen, damit es übersichtlich bleibt, wann welche Bedingung greift. 

Funktionieren tut es aber so :)

Reply:
Das `return false` ist natürlich nicht so schön, aber ich finde auch, dass es gut ist, die möglichen Werte für `sequence_mode` explizit abzuprüfen und aufzulisten.

Wir sollten aber noch einen Check einbauen, dass `sequence_mode` wirklich auf einen der drei möglichen Werte gesetzt ist. Denn sonst entsteht eine App ganz ohne Buttons ;-) Ich schaue, dass ich da gleich einen Commit anhänge, der das tut und den du dann einmal gegenchecken kannst.

Reply:
Super, mach ich :)

Ich hab grad deinen anderen Kommentar resolved, dachte man kann dann noch einen Kommentar anfügen, aber scheinbar nicht :D hab auf jeden Fall eine Erklärung für die verschiedenen Modi in die Readme gepackt.